### PR TITLE
HW-8 

### PR DESCRIPTION
**Pull request by GigiDev7**

### CODE REVIEW

**Comment by GigiDev7** [comment]
Hello, first 3 commits are Backend Stuff

**Comment by GigiDev7** [comment]
Finished HW

File: client/src/app/account-card/account-card.component.ts [hunk 1]
```diff
@@ -13,18 +14,31 @@ export class AccountCardComponent implements OnInit {
 
   public onCardClick(): void {
     this.accountService.activateAccount(this.account._id);
+    this.accountService.setAccount(this.accountService.activeAccount);
     this.transactionService.getTransactions(this.account._id).subscribe({
       next: (data) => (this.transactionService.transactions = data),
```
Comment (AnastasiaHarost) [inline client/src/app/account-card/account-card.component.ts:21]: I missed it previous time, it is bad to change value of service from component, if you need transactions here - add variable to this component

File: client/src/app/account-card/account-card.component.ts [hunk 2]
```diff
@@ -13,18 +14,31 @@ export class AccountCardComponent implements OnInit {
 
   public onCardClick(): void {
     this.accountService.activateAccount(this.account._id);
+    this.accountService.setAccount(this.accountService.activeAccount);
     this.transactionService.getTransactions(this.account._id).subscribe({
```
Comment (AnastasiaHarost) [inline client/src/app/account-card/account-card.component.ts:20]: add pipe(UntilDestroyed(this)) for all places where you subscribe on observables

File: client/src/app/account-card/account-card.component.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import { Component, Input, OnInit } from '@angular/core';
+import { InfoCardService } from '../shared/info-card/services/info-card.service';
 import { TransactionService } from '../transaction-card/services/transaction.service';
 import { AccountModel } from './account.model';
 import { AccountService } from './services/account.service';
```
Comment (AnastasiaHarost) [inline client/src/app/account-card/account-card.component.ts:5]: add @UntilDestroy before @Component

File: client/src/app/account-card/account-card.component.ts (hunk 2 above)
Comment (AnastasiaHarost) [reply client/src/app/account-card/account-card.component.ts:20]: for all places please

File: client/src/app/currencies/currency.model.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export interface CurrencyModel {
+  countryName: string;
```
Comment (AnastasiaHarost) [inline client/src/app/currencies/currency.model.ts]: let's move all interfaces to shared/model folder

File: client/src/app/main/main.component.ts [hunk 5]
```diff
@@ -16,19 +21,32 @@ export class MainComponent implements OnInit {
...
+    public transactionService: TransactionService,
+    public infoCardService: InfoCardService,
+    public formCardService: FormCardService,
+    public modalService: ModalService,
+    private currencyService: CurrencyService,
+    public notificationService: NotificationService
   ) {}
 
   public ngOnInit(): void {
     this.accountService.getAccounts().subscribe({
```
Comment (AnastasiaHarost) [inline client/src/app/main/main.component.ts:80]: pipe(UntilDestroyed(this))
@UntilDestroy
in all places, i will check)

File: client/src/app/shared/custom-button/custom-button.component.ts [hunk 6]
```diff
@@ -10,5 +11,18 @@ export class CustomButtonComponent {
   @Input() icon: string = '';
   @Input() type: string = '';
 
-  constructor() {}
+  handleClick(): void {
+    if (this.type === 'create' && this.text === 'Add Account') {
```
Comment (AnastasiaHarost) [inline client/src/app/shared/custom-button/custom-button.component.ts:15]: if (this.type === 'create') {
    if (this.text === 'Add Account') {}
    if (this.text === 'Add Transaction') {}
....
}

File: client/src/app/shared/form-card/form-card.component.ts [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+  //for category
+  public categoryType: string = '';
+
+  public categoryForm: FormGroup = new FormGroup({
+    title: new FormControl('', [Validators.required]),
+  });
+
+  public onIncomeClick(): void {
+    if (this.type === 'Category') {
```
Comment (AnastasiaHarost) [inline client/src/app/shared/form-card/form-card.component.ts:117]: this.type === 'Category' ? this.categoryType = 'income' : this.transactionType = 'income'

**Review by AnastasiaHarost — CHANGES_REQUESTED** [review]
I reviewed only FE part, if you need BE review you should add person who review BE part

**Comment by GigiDev7** [comment]
> I reviewed only FE part, if you need BE review you should add person who review BE part

yes i just mistakenly pushed it here too, i have it on another branch

**Comment by GigiDev7** [comment]
I have one question, if you know, do FE devs use next.js in exadel? or only react

**Comment by AnastasiaHarost** [comment]
> I have one question, if you know, do FE devs use next.js in exadel? or only react

I don't know to be honest, better ask Maxim Verenich

**Comment by GigiDev7** [comment]
if everything's good i'll merge it after review

File: client/src/app/shared/form-card/form-card.component.ts [hunk 8]
```diff
@@ -191,14 +221,35 @@ export class FormCardComponent {
...
     private transactionService: TransactionService,
     private reloadService: ReloadService,
     public currencyService: CurrencyService,
-    private categoryService: CategoryService,
+    public categoryService: CategoryService,
     private modalService: ModalService,
     private notificationService: NotificationService
   ) {}
+
+  ngOnInit(): void {
```
Comment (AnastasiaHarost) [inline client/src/app/shared/form-card/form-card.component.ts:270]: public

File: client/src/app/shared/form-card/form-card.component.ts [hunk 9]
```diff
@@ -191,14 +221,35 @@ export class FormCardComponent {
...
+    this.categoryService.getCategories().subscribe({
```
Comment (AnastasiaHarost) [inline client/src/app/shared/form-card/form-card.component.ts]: untilDestroyed

File: client/src/app/shared/form-card/form-card.component.ts [hunk 10]
```diff
@@ -209,7 +228,11 @@ export class FormCardComponent implements OnInit {
     } else if (this.type === 'Category') {
       const { title } = this.categoryForm.value;
       this.categoryService.addCategory(this.categoryType, title).subscribe({
```
Comment (AnastasiaHarost) [inline client/src/app/shared/form-card/form-card.component.ts:230]: use switchMap

File: client/src/app/shared/modal-card/modal-card.component.ts [hunk 11]
```diff
@@ -28,14 +30,21 @@ export class ModalCardComponent {
...
+      this.accountService
+        .deleteAccount(account._id)
+        .pipe(untilDestroyed(this))
+        .subscribe();
       this.reloadService.reloadComponent();
       this.notificationService.showNotification();
       this.notificationService.setNotificationText('Account Deleted');
     } else {
       const transaction = this.transactionService.singleTransaction;
       this.transactionService.deleteTransaction(transaction._id).subscribe({
```
Comment (AnastasiaHarost) [inline client/src/app/shared/modal-card/modal-card.component.ts:42]: switchMAp

File: client/src/app/shared/form-card/form-card.component.ts (hunk 9 above)
Comment (GigiDev7) [reply client/src/app/shared/form-card/form-card.component.ts]: its updated in new commits, as you said i did not left single subscribtion without untilDestroy :d

**Pull request merged**